clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -58,6 +58,8 @@ enum KernelInvocationKind {
 
 const static std::string InitMethodName = "__init";
 const static std::string InitESIMDMethodName = "__init_esimd";
+const static std::string InitSpecConstantsBuffer =
```

Comment:
This annoys the crap outa me.  I realize you're picking up the current pattern, but these don't need to be std::string.  These should all be constexpr StringLiterals: https://llvm.org/doxygen/classllvm_1_1StringLiteral.html

Reply:
FYI, i'll just push something to fix the surrounding cases.